src/libsyntax/util/parser.rs

```diff
@@ -324,6 +323,8 @@ impl ExprPrecedence {
             // Unary, prefix
             ExprPrecedence::Box |
             ExprPrecedence::AddrOf |
+            // Here `let pats = expr` has `let pats =` as a "unary" prefix of `expr`.
+            ExprPrecedence::Let |
```

Comment:
Pretty-printing from AST/HIR currently works correctly even in absence of parens (thanks to great work in https://github.com/rust-lang/rust/pull/43742).

I suspect that the "unary" precedence here is incorrect and will cause pretty-printing for `let` work incorrectly if it has low-priority operators inside it and high priority outside, like `(let a = b || c) && d`.

This is probably not a big deal, but you can try to add a test case for this to `run-pass-fulldeps/pprust-expr-roundtrip.rs`.

Reply:
Fixed (implementation and test).